ComponentKit/Core/Scope/CKScopedComponentController.h

```diff
@@ -0,0 +1,17 @@
+//
+//  CKScopedComponentController.h
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/27/17.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+@protocol CKScopedComponent;
+
+@protocol CKScopedComponentController <NSObject>
+
+- (instancetype)initWithComponent:(id<CKScopedComponent>)component;
```

Comment:
The scope infra actually constructs component controllers at the moment. I considered pulling this out and passing in a component controller factory, but decided against for now. May need that for section components though.